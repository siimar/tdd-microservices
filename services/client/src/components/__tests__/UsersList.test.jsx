import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import UsersList from '../UsersList';

const users = [
  {
    'active': true,
    'email': 'test@foo.ee',
    'id': 1,
    'username': 'test'
  },
  {
    'active': true,
    'email': 'test2@foo.ee',
    'id': 2,
    'username': 'test2'
  }
];

test('UsersList renders properly', () => {
  const wrapper = shallow(<UsersList users={users}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('test');
});

test('UsersList renders a snapshot properly', () => {
  const tree = renderer.create(<UsersList users={users}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
