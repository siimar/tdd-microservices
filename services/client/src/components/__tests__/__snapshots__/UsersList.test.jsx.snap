// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h1
    className="title is-1"
  >
    All Users
  </h1>
  <hr />
  <br />
  <table
    className="table is-hoverable is-fullwidth"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Email
        </th>
        <th>
          Username
        </th>
        <th>
          Active
        </th>
        <th>
          Admin
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          test@foo.ee
        </td>
        <td>
          test
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          test2@foo.ee
        </td>
        <td>
          test2
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
